test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,8 +92,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
+        Command command = new Parser().parseCommand(inputCommand);
+        if(command.isMutating()){
         assertEquals(addressBook, saveFile.load());
```

Comment:
Good job for updating the test case. However, there is a coding standard violation and indentation problem.